enterprise/frontend/src/metabase-enterprise/metabot/components/MetabotAdmin/MetabotPurchasePage.tsx

```diff
@@ -0,0 +1,200 @@
+import { useDisclosure } from "@mantine/hooks";
+import type { FetchBaseQueryError } from "@reduxjs/toolkit/query";
+import { useCallback } from "react";
+import { t } from "ttag";
+import * as Yup from "yup";
+
+import {
+  SettingsPageWrapper,
+  SettingsSection,
+} from "metabase/admin/components/SettingsSection";
+import { getCurrentUser } from "metabase/admin/datamodel/selectors";
+import { usePurchaseMetabotCloudAddOnMutation } from "metabase/api/metabot";
+import ExternalLink from "metabase/common/components/ExternalLink";
+import { useSetting } from "metabase/common/hooks";
+import {
+  Form,
+  FormCheckbox,
+  FormProvider,
+  FormSubmitButton,
+} from "metabase/forms";
+import { useSelector } from "metabase/lib/redux";
+import { Divider, Flex, Group, List, Stack, Text } from "metabase/ui";
+import { MetabotPurchaseSettingUpModal } from "metabase-enterprise/metabot/components/MetabotAdmin/MetabotPurchaseSettingUpModal";
+
+// https://redux-toolkit.js.org/rtk-query/usage/error-handling
+// https://redux-toolkit.js.org/rtk-query/usage-with-typescript#type-safe-error-handling
+const isFetchBaseQueryError = (error: unknown): error is FetchBaseQueryError =>
+  error instanceof Object && "status" in error && "data" in error;
+
+type IFieldError =
+  | string
+  | {
+      message: string;
+    }
+  | {
+      errors: { [key: string]: any };
+    };
+
+const isFieldError = (error: unknown): error is IFieldError =>
+  typeof error === "string" ||
+  (error instanceof Object &&
+    (("message" in error && typeof error.message === "string") ||
+      ("errors" in error &&
+        error.errors instanceof Object &&
+        "terms_of_service" in error.errors &&
+        typeof error.errors.terms_of_service === "string")));
+
+export const handleFieldError = (error: unknown) => {
+  if (!isFieldError(error)) {
+    return;
+  }
+
+  if (typeof error === "string") {
+    throw { data: { errors: { terms_of_service: error } } };
+  }
+
+  if ("message" in error) {
+    throw { data: { errors: { terms_of_service: error.message } } };
+  }
+
+  if ("errors" in error) {
+    throw { data: error };
+  }
+};
+
+const validationSchema = Yup.object({
+  terms_of_service: Yup.boolean(),
+});
+
+interface MetabotPurchaseFormFields {
+  terms_of_service: boolean;
+}
+
+export const MetabotPurchasePage = () => {
+  const tokenStatus = useSetting("token-status");
+  const currentUser = useSelector(getCurrentUser);
+  const isStoreUser =
+    tokenStatus?.["store-users"]
+      ?.map(({ email }) => email)
+      ?.includes(currentUser?.email) ?? false;
+
+  const [settingUpModalOpened, settingUpModalHandlers] = useDisclosure(false);
+  const [purchaseMetabotAddOn] = usePurchaseMetabotCloudAddOnMutation();
+  const onSubmit = useCallback(
+    async ({ terms_of_service }: MetabotPurchaseFormFields) => {
+      await purchaseMetabotAddOn({
```

Comment:
When I was testing this I had to wait a bit before modal opened, and when it opened then Metabot was immediately enabled. Aren't we waiting [for both enabling the Metabot AI and refreshing the token](https://github.com/metabase/metabase/pull/62513/files#diff-ca407010a35905a2fcfbccf6a0900b8f89d9a045b12ee7216c4eab08a9383e98R38). Maybe we should open the modal right away and present the loading state in modal, instead of showing the blocked button for extended time.

Reply:
Should we then do something about the "Success" text on the `FormSubmitButton`?  It appears a bit odd to me to have the form greyed out and covered by a modal that says "wait" and in the background see a green "Success" light up.

Reply:
> Aren't we waiting for both enabling the Metabot AI and refreshing the token.

We wait only for purchasing the add-on and dropping the token cache.  Then the frontend triggers another token check when showing the modal: https://github.com/metabase/metabase/pull/62513/files#diff-c657ec9cd4c5e1b8fa7ec286927a67fe2421a010bcfae333c036c705b5db3e12R23

Reply:
Showing the modal earlier: https://github.com/metabase/metabase/pull/62513/commits/31ddb144831b0fd9b037504388c8a8a68671de1d

Reply:
Nice! Better UX ☺️